### DISCUSSION TITLE
sensor stuck accepting salt keys

### DISCUSSION THREAD

**Discussion by SanibelJack** [Security-Onion-Solutions/securityonion#3752] [Unsupported Versions]
Hello everyone,
I'm installing my first sensor on my new 2.3 install but it is stuck at 20% with "Accepting Salt Key on manager". Anyone have any suggestions?

Thanks 
Jack

**Comment by palichx** [comment]
Hi,
check manager hostname and soremote password.

**Reply by defensivedepth** [reply]
Also check the setup log at `/root/sosetup.log`

**Reply by ranguli** [reply]
I'm also having this issue as mentioned above. I can verify that the manager hostname and soremote password are correct. The output from `/root/sosetup.log` is:

```
Executing command with retry support: accept_salt_key_remote
Accept the key remotely on the manager
Results: The key glob 'security-onion-sensor_sensor' does not match any unaccepted keys.
The key glob 'security-onion-sensor_sensor' does not match any unaccepted keys. (0)
Expected 'going to be accepted' but got 'The key glob 'security-onion-sensor_sensor' does not match any unaccepted keys.
The key glob 'security-onion-sensor_sensor' does not match any unaccepted keys.'
Command failed with exit code 0; will retry in 10 seconds (1 / 20)...
```

Additionally the output of `salt-key` over on the manager node:
```
[onion@security-onion-manager ~]$ sudo salt-key
Accepted Keys:
security-onion-manager_manager
Denied Keys:
Unaccepted Keys:
Rejected Keys:
```

Any help would be greatly appreciated!

**Comment by SanibelJack** [comment]
Thank you I’ll look

Jack B. Voth

Sent from my iPhone.  Please excuse the brevity, spelling and punctuation.

On Apr 5, 2021, at 1:38 PM, Josh Brower ***@***.***> wrote:

﻿
Also check the setup log at /root/sosetup.log —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌  ‌
Neutral (From: ***@***.******@***.***>, External)
Report This Email<https://shared.outlook.inky.com/report?id=c2FuaWJlbGN5YmVydGVjaC9qYWNrLnZvdGhAc2FuaWJlbGN5YmVydGVjaC5jb20vYWZiNDM3NzNkY2ZjOGNkMGM4OTQxODA2ZjIzMDRhYmMvMTYxNzY0NDI4OS45MQ==#key=3ab3c5c3d299aba42e813ef7d825bf07>  FAQ<https://www.inky.com/banner-faq/>  Protection by INKY<https://www.inky.com>


Also check the setup log at /root/sosetup.log

**Comment by hacker0ni** [comment]
Hey not sure if this will still help but it may help others who had the same issue. This same issue happened to me because I tried redeploying a sensor on a new Grid. I've removed salt packages, everything under `/opt/so/` and re-ran the installer. The issue was making sure the firewall allowed access to the "new" forward node. By default sensors should be in `sensor` and `minion` portgroups in the firewall. You can do this by executing the following commands on your manager:
```
so-firewall includehost sensor <SENSOR_IP_HERE>
so-firewall includehost minion <SENSOR_IP_HERE>
so-firewall apply
```

**Reply by dougburks** [reply]
In theory, the manager should automatically create the necessary firewall entries when the new sensor joins the grid.

**Reply by 2manyvcos** [reply]
I have the same problem whenever I try to join a sensor with version v2.3.120 to a manager with version v2.3.130. (I prepared multiple servers by installing the ISO on them so that I only need to configure them when we need one.)

Adding them manually to the firewall like how @hacker0ni described fixes the problem.

**Reply by dougburks** [reply]
@2manyvcos We generally recommend installing sensors with the same version as the manager.
